File: app/src/main/java/com/dhsdevelopments/potato/clientapi/message/TypedMessageElement.java
package com.dhsdevelopments.potato.clientapi.message;

public class TypedMessageElement extends MessageElement
{
    protected MessageElement content;

    public TypedMessageElement( MessageElement content ) {
        this.content = content;
    }

    @Override
    public String toString() {
        return "TypedMessageElement[type=" + getClass().getName() +
                       ", content=" + content +
                       "] " + super.toString();
    }
}

File: app/src/main/java/com/dhsdevelopments/potato/clientapi/message/TypedMessageElement.kt
package com.dhsdevelopments.potato.clientapi.message

open class TypedMessageElement(protected var content: MessageElement) : MessageElement() {

    override fun toString(): String {
        return "TypedMessageElement[type=" + javaClass.name +
                ", content=" + content +
                "] " + super.toString()
    }
}
